src/imagecapturecore.cs

```diff
@@ -0,0 +1,900 @@
+	interface ICCameraDeviceDownloadDelegate {
+
+		[Export ("didDownloadFile:error:options:contextInfo:")]
+		void DidDownloadFile (ICCameraFile file, [NullAllowed] NSError error, [NullAllowed] NSDictionary<NSString, NSObject> options, [NullAllowed] IntPtr contextInfo);
+
+		[Export ("didReceiveDownloadProgressForFile:downloadedBytes:maxBytes:")]
+		void DidReceiveDownloadProgress (ICCameraFile file, long downloadedBytes, long maxBytes);
+	}
+
+	[BaseType (typeof(ICDevice))]
+	[DisableDefaultCtor] // ICDeviceBrowser creates instances of this class
+	interface ICCameraDevice {
+
+		[Export ("batteryLevelAvailable")]
+		bool BatteryLevelAvailable { get; }
+
+		[Export ("batteryLevel")]
+		nuint BatteryLevel { get; }
+
+		[Export ("contentCatalogPercentCompleted")]
+		nuint ContentCatalogPercentCompleted { get; }
+
+		[NullAllowed, Export ("contents")]
+		ICCameraItem[] Contents { get; }
+
+		[NullAllowed, Export ("mediaFiles")]
+		ICCameraItem[] MediaFiles { get; }
+
+		[Export ("timeOffset")]
+		double TimeOffset { get; }
+
+		[Export ("isAccessRestrictedAppleDevice")]
+		bool IsAccessRestrictedAppleDevice { get; }
+
+		[NullAllowed, Export ("mountPoint")]
+		string MountPoint { get; }
+
+		[Export ("tetheredCaptureEnabled")]
+		bool TetheredCaptureEnabled { get; }
+
+		[Export ("filesOfType:")]
+		[return: NullAllowed]
+		string[] GetFiles (string fileUTType);
+
+		[Export ("requestSyncClock")]
+		void RequestSyncClock ();
+
+		[Export ("requestEnableTethering")]
+		void RequestEnableTethering ();
+
+		[Export ("requestDisableTethering")]
+		void RequestDisableTethering ();
+
+		[Export ("requestTakePicture")]
+		void RequestTakePicture ();
+
+		[Export ("requestDeleteFiles:")]
+		void RequestDeleteFiles (ICCameraItem[] files);
+
+		[Export ("cancelDelete")]
+		void CancelDelete ();
+
+		[Export ("requestDownloadFile:options:downloadDelegate:didDownloadSelector:contextInfo:")]
+		void RequestDownloadFile (ICCameraFile file, [NullAllowed] NSDictionary<NSString, NSObject> options, IICCameraDeviceDownloadDelegate downloadDelegate, Selector didDownloadSelector, [NullAllowed] IntPtr contextInfo);
+
+		[Export ("cancelDownload")]
+		void CancelDownload ();
+
+		[Export ("requestUploadFile:options:uploadDelegate:didUploadSelector:contextInfo:")]
+		void RequestUploadFile (NSUrl fileUrl, [NullAllowed] NSDictionary<NSString, NSObject> options, NSObject uploadDelegate, Selector didUploadSelector, [NullAllowed] IntPtr contextInfo);
+
+		[Export ("requestReadDataFromFile:atOffset:length:readDelegate:didReadDataSelector:contextInfo:")]
+		void RequestReadDataFromFile (ICCameraFile file, long offset, long length, NSObject readDelegate, Selector didReadDataSelector, [NullAllowed] IntPtr contextInfo);
+
+		[Export ("requestSendPTPCommand:outData:sendCommandDelegate:didSendCommandSelector:contextInfo:")]
+		void RequestSendPtpCommand (NSData command, NSData outData, NSObject sendCommandDelegate, Selector didSendCommandSelector, [NullAllowed] IntPtr contextInfo);
```

Comment:
For the Request* methods that take a Delegate + Selector + Context Info we might want to provide a different overload that takes a single delegate parameter instead:

```csharp
delegate void DidSendPtpCommand (NSData command, NSData inData, NSData response, NSError error);
public void RequestSendPtpCommand (NSData command, NSData outData, DidSendPtpCommand callback)
{
    // Magic goes here
}
```

Reply:
@dalexsoto is that API used in your sample ?

It's nice to hide (or provide better API) than exposing `Selector` and `NSObject` in signatures - but without tests it's also easy too easy to get it wrong.

Since it's not a very common API (or framework) on macOS (only) I think it's fine to file an enhancement on it (would be a nice and not _too_ hard contribution from anyone consuming the API).

Reply:
@spouliot Unfortunately no, it does not touch any of those APIs, I have opened this issue https://github.com/xamarin/xamarin-macios/issues/5093 per your request